src/lib.rs

```diff
@@ -57,54 +57,108 @@ fn validate(payload: &[u8]) -> CallResult {
+    #[case(None, HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),Some("no-priority".to_string()), true)]
+    #[case(Some("low-priority".to_string()), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),HashSet::new(), Some("no-priority".to_string()), true)]
+    #[case(Some("other-priority".to_string()), HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),Some("no-priority".to_string()), true)]
+    #[case(Some("low-priority".to_string()), HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]), Some("no-priority".to_string()), false)]
+    #[case(Some("no-priority".to_string()), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),HashSet::new(), Some("no-priority".to_string()), false)]
     fn test_pod_validation(
         #[case] pod_priority_class: Option<String>,
         #[case] allowed_classes: HashSet<String>,
         #[case] denied_classes: HashSet<String>,
+        #[case] default_priority_class: Option<String>,
         #[case] should_succeed: bool,
     ) {
-        let pod = PodSpec {
-            priority_class_name: pod_priority_class,
+        let mut pod = PodSpec {
+            priority_class_name: pod_priority_class.clone(),
             ..Default::default()
         };
-        let result = validate_pod_priority_class(pod, allowed_classes, denied_classes);
+        let settings = Settings {
+            allowed_priority_classes: allowed_classes,
+            denied_priority_classes: denied_classes,
+            default_priority_class: default_priority_class.clone(),
+        };
+        let result = validate_pod_priority_class(&mut pod, &settings);
         assert_eq!(result.is_ok(), should_succeed);
+        match (
+            should_succeed,
+            default_priority_class.clone(),
+            pod_priority_class.clone(),
+        ) {
+            (true, Some(_), None) => {
+                let mutated = result.expect("Should be Ok");
+                assert_eq!(mutated, POD_SPEC_MUTATE);
+                assert_eq!(pod.priority_class_name, default_priority_class);
+            }
+            (true, Some(_), Some(_)) => {
+                let mutated = result.expect("Should be Ok");
+                assert_eq!(mutated, POD_SPEC_NOT_MUTATE);
+                assert_eq!(pod.priority_class_name, pod_priority_class);
+            }
+            (true, None, Some(_)) => {
+                let mutated = result.expect("Should be Ok");
+                assert_eq!(mutated, POD_SPEC_NOT_MUTATE);
+                assert_eq!(pod.priority_class_name, pod_priority_class);
+            }
+            (true, None, None) => {
+                let mutated = result.expect("Should be Ok");
+                assert_eq!(mutated, POD_SPEC_NOT_MUTATE);
+                assert_eq!(pod.priority_class_name, pod_priority_class);
+            }
+            (false, _, _) => (),
+        }
```

Comment:
Some comments, I think you can leave things as they are, but in the future you can also consider writing another test. For example, one that tests validation (with no mutation) and one that tests mutation.

I would change the current code to get rid of the `success` boolean from the `match`.
Actually, we're already checking that at line 135.

I propose to do something like that:

```rust
let result = validate_pod_priority_class(&mut pod, &settings);
assert_eq!(result.is_ok(), should_succeed);
if !should_succeed {
  return
}
let has_been_mutated = result.unwrap();
match ...
```

```